src/App.tsx

```diff
@@ -1,7 +1,47 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type Props = {};
+
+type State = {
+  key: string;
+};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    key: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.handleKeyEvent);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.handleKeyEvent);
+  }
+
+  handleKeyEvent = (event: KeyboardEvent) => {
+    this.setState({
+      key: event.key,
+    });
+
+    if (event.key === ' ') {
+      this.setState({
+        key: 'Space',
+      });
+    }
+  };
```

Comment:
I didn't understand what this if statement does and couldn't test because of the npm test error. Can you help me to understand?